incubator/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/stream/MqttKafkaPublishFactory.java

```diff
@@ -206,6 +202,8 @@ private MqttPublishProxy(
             this.replyId = supplyReplyId.applyAsLong(initialId);
             this.messages = new KafkaMessagesProxy(originId, resolvedId, this);
             this.retained = new KafkaRetainedProxy(originId, resolvedId, this);
+            this.kafkaMessagesTopic = kafkaMessagesTopic;
+            this.kafkaRetainedTopic = kafkaRetainedTopic;
```

Comment:
Both `kafkaMessagesTopic` and `kafkaRetainedTopic` should probably be `String16FW` type instead, per comments above.